### PR TITLE
added deprecated warning for ctype solar-x/solar-y conversion

### PR DESCRIPTION
**Pull request by jayantbhakar**
## PR Description

A deprecated warning message has been added to the mapbase. When the ctype solar-x/solar-y is changed to 'HPLN-TAN/HPLT-TAN', a warning message will be shown telling the feature is deprecated.
It will also help in discovering other map source dependents of this conversion, which can be fixed individually.

This PR is based on issue #5569.

### CODE REVIEW

**Comment by jayantbhakar** [comment]
pre-commit.ci autofix

File: changelog/6901.bugfix.rst [hunk 1]
```diff
@@ -0,0 +1,2 @@
+added a deprecated warning message when the ``ctype solar-x/solar-y`` conversion happens in `sunpy/map/mapbase.py`.
+It will also help in discovering other map source dependents of this conversion, which can be fixed.
```
Comment (nabobalis) [inline changelog/6901.bugfix.rst:2]:
```suggestion
We have deprecated the conversion of WCS ``ctype `` values of "solar-x", "solar_x" within `~sunpy.map.Map`.
```

**Comment by nabobalis** [comment]
Looks like we have test fails so we need to work out what is triggering this and fix it internally.

File: changelog/6901.bugfix.rst
[the remark this answers is not in the document]
Comment (jayantbhakar) [reply changelog/6901.bugfix.rst:1]: how about depreciated?

**Comment by jayantbhakar** [comment]
The tests are failing due to the deprecated warning I have added.
In the figure tests the following warning shows up
`
../../.tox/py310-figure/lib/python3.10/site-packages/sunpy/util/exceptions.py:105 SunpyDeprecationWarning
sunpy.util.exceptions.SunpyDeprecationWarning: The ctype1 solar-x/solar_x conversion is deprecated and will be removed in the future release.
`
How do we fix this?

**Comment by nabobalis** [comment]
We will have to catch it with pytest. See https://docs.pytest.org/en/7.2.x/how-to/capture-warnings.html

**Comment by jayantbhakar** [comment]
pre-commit.ci autofix

**Comment by jayantbhakar** [comment]
I have captured the warnings in the individual test files and tests are passing now.

**Comment by nabobalis** [comment]
Actually I was wrong. We do not want to filter the warnings like this.

What we want to do is for each test that is failing, find the data file causing that problem. From there, find the Map source class under [`sources`](https://github.com/sunpy/sunpy/tree/main/sunpy/map/sources), then add the ctype solar-x/solar-y if statement, so not to raise the deprecation warning that is now raised by the core Map class.

**Pull request closed without merging**